cns/middlewares/k8sSwiftV2_linux.go

```diff
@@ -0,0 +1,124 @@
+			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
+		}
+		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
+		}
+
+		// Get and parse podCIDRs from env
+		podCIDRs, err := configuration.PodCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get podCIDRs from env")
+		}
+		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse podCIDRs")
+		}
+
+		// Get and parse serviceCIDRs from env
+		serviceCIDRs, err := configuration.ServiceCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get serviceCIDRs from env")
+		}
+		serviceCIDRsV4, serviceCIDRsV6, err := utils.ParseCIDRs(serviceCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse serviceCIDRs")
+		}
+		// Check if the podIPInfo is IPv4 or IPv6
+		ip, err := netip.ParseAddr(podIPInfo.PodIPConfig.IPAddress)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse podIPConfig IP address %s", podIPInfo.PodIPConfig.IPAddress)
+		}
+		if ip.Is4() {
+			// routes for IPv4 podCIDR traffic
+			for _, podCIDRv4 := range podCIDRsV4 {
+				podCIDRv4Route := cns.Route{
+					IPAddress:        podCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv4Route)
+			}
+			// route for IPv4 serviceCIDR traffic
+			for _, serviceCIDRv4 := range serviceCIDRsV4 {
+				serviceCIDRv4Route := cns.Route{
+					IPAddress:        serviceCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv4Route)
+			}
+			// route for IPv4 infraVNETCIDR traffic
+			for _, infraVNETCIDRv4 := range infraVNETCIDRsv4 {
+				infraVNETCIDRv4Route := cns.Route{
+					IPAddress:        infraVNETCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, infraVNETCIDRv4Route)
+			}
+		} else {
+			// routes for IPv6 podCIDR traffic
+			for _, podCIDRv6 := range podCIDRv6 {
+				podCIDRv6Route := cns.Route{
+					IPAddress:        podCIDRv6,
+					GatewayIPAddress: overlayGatewayV6,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv6Route)
+			}
+			// route for IPv6 serviceCIDR traffic
+			for _, serviceCIDRv6 := range serviceCIDRsV6 {
+				serviceCIDRv6Route := cns.Route{
+					IPAddress:        serviceCIDRv6,
+					GatewayIPAddress: overlayGatewayV6,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv6Route)
+			}
+			// route for IPv6 infraVNETCIDR traffic
+			for _, infraVNETCIDRv6 := range infraVNETCIDRsv6 {
+				infraVNETCIDRv6Route := cns.Route{
+					IPAddress:        infraVNETCIDRv6,
+					GatewayIPAddress: overlayGatewayV6,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, infraVNETCIDRv6Route)
+			}
+		}
+		podIPInfo.SkipDefaultRoutes = true
+	case cns.BackendNIC:
+	case cns.AccelnetNIC:
```

Comment:
These cases do not fall through btw... it will return nil. Is that the intention here?

Reply:
It is intentionally left blank for future development.

Reply:
Might be worth a comment to that effect?

Reply:
Added comments.